pkg/virt-controller/watch/snapshot/restore.go

```diff
@@ -565,9 +569,46 @@ func (t *vmRestoreTarget) Reconcile() (bool, error) {
 	}
 	t.vm = newVM
 
+	for _, dvt := range newTemplates {
+		if err := t.createDataVolume(dvt); err != nil {
+			return false, err
+		}
+	}
+
 	return true, nil
 }
 
+func (t *vmRestoreTarget) createDataVolume(dvt kubevirtv1.DataVolumeTemplateSpec) error {
+	newDataVolume, err := watchutil.CreateDataVolumeManifest(t.controller.Client, dvt, t.vm)
+	if err != nil {
+		return fmt.Errorf("Unable to create restore DataVolume manifest: %v", err)
+	}
+
+	pvcName := ""
+	for _, vr := range t.vmRestore.Status.Restores {
+		if dvName := vr.DataVolumeName; dvName != nil && *dvName == newDataVolume.Name {
+			pvcName = vr.PersistentVolumeClaimName
+			break
+		}
+	}
+	if pvcName == "" {
+		return fmt.Errorf("No VolumeRestore found for DataVolume %s", newDataVolume.Name)
+	}
+	pvc, err := t.controller.getPVC(t.vm.Namespace, pvcName)
+	if err != nil {
+		return err
+	}
+	if pvc.Annotations[populatedForPVCAnnotation] != dvt.Name || len(pvc.OwnerReferences) > 0 {
+		return nil
+	}
+
+	if _, err = t.controller.Client.CdiClient().CdiV1beta1().DataVolumes(t.vm.Namespace).Create(context.Background(), newDataVolume, v1.CreateOptions{}); err != nil {
```

Comment:
this should be done similar to reconcile reconcileVolumeRestores -> wait until datavolume completed

Reply:
also like done in handleDataVolumes -> not ready until exists and status is completed

Reply:
done

Reply:
Now that I think about it, it's actually a bug that we wait for restore PVCs to be bound during restore.  What if they are WaitForFirsConsumer?  Nevertheless, the DV *should* be Succeeded if the PVC is "populatedFor" and WFFC (the case here).  If not, that's another bug.